tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,706 @@
+            if let AuthorizedKeyEntry::Valid { ref key } = key {
+                assert!(key
+                    .to_fingerprint_string()
+                    .contains(&*FINGERPRINTS[sshkeyname2]));
+
+                aks.write()
+                    .expect("failed to update authorized keys directory");
+                aks.sync().expect("failed to update authorized keys");
+
+                let authkeyone: PathBuf =
+                    PathBuf::from(format!("{}/authorized_keys.d/{}", ssh_dir, "two"));
+                assert!(authkeyone.exists());
+
+                let mut svec: Vec<&str> = Vec::new();
+                svec.push(sshkeyname1);
+                assert_has_keys(ssh_dir, svec, true);
+            }
+        }
+
+        // add two again
+        // "update-ssh-keys --add two valid2.pub"
+        //
+        // NOTE: Ideally adding two again here is supposed to immediately fail,
+        // but for some reason it simply hangs forever. For now do not run the
+        // following command to proceed to the next tests.
+
+        //        let assert_keys = ["valid1"].to_vec();
+        //        match add_one_ssh_key(
+        //            ssh_dir,      // ssh_dir
+        //            pubkeyvalid2, // pubkeyname
+        //            testnametwo,  // testname
+        //            false,        // is_force
+        //            true,         // is_replace
+        //        ) {
+        //            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnametwo, assert_keys),
+        //            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        //        }
+    }
+
+    #[test]
+    fn test_disable() {
+        let ctx = setup_tests();
+
+        run_test_disable(&ctx.ssh_dir);
+    }
+
+    #[test]
+    fn test_enable() {
+        let ctx = setup_tests();
+
+        let ssh_dir_path = &ctx.ssh_dir;
+        let ssh_dir = ssh_dir_path.to_str().unwrap();
+        let pubkeyvalid2 = "valid2";
+        let testnametwo = "two";
+        let assert_keys = ["valid1", "valid2"].to_vec();
+
+        // "update-ssh-keys --disable two"
+        run_test_disable(ssh_dir_path);
+
+        // "update-ssh-keys --add-force two valid2.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid2, // pubkeyname
+            testnametwo,  // testname
+            true,         // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => {
+                add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnametwo, assert_keys)
+            }
+            Err(err) => panic!("update_ssh_keys --add-force failed {}", err),
+        }
+    }
+
+    #[test]
+    fn test_add_bad() {
+        let ctx = setup_tests();
+
+        let ssh_dir_path = &ctx.ssh_dir;
+        let ssh_dir = ssh_dir_path.to_str().unwrap();
+        let pubkeybad = "bad";
+        let testnamebad = "bad";
+        let assert_keys = ["bad"].to_vec();
```

Comment:
```suggestion
        let assert_keys = ["valid1"].to_vec();
```
The Python test has `self.assertHasKeys('valid1')`

Reply:
Done